File: pharmacy-backend/src/main/java/com/harshit/pharmacy/order/service/AdminOrderService.java
package com.harshit.pharmacy.order.service;

import com.harshit.pharmacy.order.dto.OrderReportResponse;
import com.harshit.pharmacy.order.dto.OrderResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;

public interface AdminOrderService {


    Page<OrderResponse> getOrders(String status, Pageable pageable);

    void updatePrescriptionStatus(Integer orderId, String status);

    void changeOrderStatus(Integer orderId, String status);

    void cancelOrder(Integer orderId);

    OrderResponse getOrderById(Integer orderId);

    Page<OrderReportResponse> getOrderReport(LocalDate startDate, LocalDate endDate, Pageable pageable);

}

File: pharmacy-frontend/src/services/adminOrderService.js
import api from "../api/api";

export async function getOrders(params) {
    const response = await api.get("/admin/orders", { params });
    return response.data;
}

export async function updatePrescriptionStatus(orderId, prescriptionStatus) {
    const response = await api.patch(`/admin/orders/${orderId}/prescription`, { prescriptionStatus });
    return response.data;
}

export async function updateOrderStatus(orderId, orderStatus) {
    const response = await api.patch(`/admin/orders/${orderId}/status`, { orderStatus });
    return response.data;
}

export async function cancelOrder(orderId) {
    const response = await api.patch(`/admin/orders/${orderId}/cancel`);
    return response.data;
}

export async function getOrderById(orderId) {
    const response = await api.get(`/admin/orders/${orderId}`);
    return response.data;
}

export async function getOrderReport({ startDate, endDate, page, size }) {
    const response = await api.get("/admin/orders/report", {
        params: { startDate, endDate, page, size }
    });
    return response.data;
}
